From f6c6c0bfbf9986cad974add6698ee9a2912b6a07 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 24 Dec 2014 16:51:22 +0800
Subject: [PATCH 0331/5242] MLK-10055-2: mfd: si476x-i2c: sound is registered
 when no FM module attached

commit  89b92ae5e6183d815545fa698fffb9747730f0fc from
https://source.codeaurora.org/external/imx/linux-imx.git

The si476x_core_get_revision_info will send i2c command to FM module, if it
return error, there is no FM modules attached, so we need't to register the
sound card. otherwise, the pulseaudio will access this sound card, but return
a lot of i2c error.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 4c8e9916128f05f9b4115e1ee1af4a1e7d800c4a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/si476x-i2c.c |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/mfd/si476x-i2c.c b/drivers/mfd/si476x-i2c.c
index b75594b..75b6c44 100644
--- a/drivers/mfd/si476x-i2c.c
+++ b/drivers/mfd/si476x-i2c.c
@@ -809,15 +809,14 @@ static int si476x_core_probe(struct i2c_client *client,
 	core->power_state = SI476X_POWER_UP_FULL;
 	si476x_core_set_power_state(core, SI476X_POWER_DOWN);
 
+	rval = si476x_core_get_revision_info(core);
+	if (rval < 0) {
+		rval = -ENODEV;
+		goto free_kfifo;
+	}
+
 	if (of_property_read_bool(client->dev.of_node, "revision-a10"))
 		core->revision = SI476X_REVISION_A10;
-	else {
-		rval = si476x_core_get_revision_info(core);
-		if (rval < 0) {
-			rval = -ENODEV;
-			goto free_kfifo;
-		}
-	}
 
 	cell_num = 0;
 
-- 
1.7.9.5

